Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/SingleFileTransfer.cs

```diff
@@ -47,18 +47,20 @@ public SingleFileTransfer(FileItem file, string destinationDirectory, BatchConte
     /// </summary>
     internal TransferState State { get; set; }
 
-    internal SingleFileTransferResult TransferSingleFile()
+    internal SingleFileTransferResult TransferSingleFile(CancellationToken cancellationToken)
     {
         try
         {
             var test = DestinationFileWithMacrosExpanded;
 
+            cancellationToken.ThrowIfCancellationRequested();

```

Comment:
No need to check cancellation here.

Reply:
The Task has been known for getting stuck and cannot be terminated. This is the reason I added these cancellation checks in different layers of the Task.